taskcluster/scriptworker_taskgraph/transforms/cached.py

```diff
@@ -0,0 +1,95 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
Hm, I didn't stumble upon this type of transform in neither of the mobile projects nor in-tree gecko. Maybe worth adding a short comment on it?

Reply:
Sure, I'll add a comment. It's similar to https://searchfox.org/mozilla-central/source/taskcluster/taskgraph/transforms/cached_tasks.py . I use it in xpi and adhoc; every taskgraph repo uses cached tasks for docker images and fetches... this is what keeps tasks from running if the underlying code hasn't changed.

Reply:
I was under the impression that it's based on the routes. Like using taskgraph in a repo generically sets specific routes, which are downstream used in taskgraph to find previously-run tasks. E.g. https://hg.mozilla.org/ci/taskgraph/file/tip/src/taskgraph/util/cached_tasks.py#l20 - but looking at that code again, maybe we use that only for toolchain tasks or so? 

Ah, looking at the URl you pointed, seems like that's used under the hood as well - https://searchfox.org/mozilla-central/source/taskcluster/taskgraph/transforms/cached_tasks.py#75 - I wonder why we never added this transform in any of the mobile projects. I'll dig further.

Reply:
`taskgraph.transfrorms.cached_task` is a transform that takes a `cache` value on a task (with cache type, name and digest data), and turns it into the appropriate routes and optimizations so that the task will be re-used as long as the digest data is not changed.

That transform is used by the docker-image, toolchain and fetch transforms to implement their caching behavior. This transform is similar to those, in that it generates a `cache` value with appropriate digest to be consumed by `taskgraph.transforms.cached_task`; but unlike them, does not expect the task to have any particular artifacts, and doesn't have any other bits of taskgraph that are designed to interact with it.

> I wonder why we never added this transform in any of the mobile projects. I'll dig further.

It is not obvious to me where it would be useful for mobile projects. The most likely would be android-components or application-services. Though, given that android-components has a unified build system, I suspect it would be somewhat less useful for that project.

To a certain extent, this transform is being used to work-around the fact that we don't have a push-log for this repo, which would let us see if any relevant files changed since the last push. On the other hand, this precise tracking, as backing something out will not rerun tasks on the old code.